src/msg/async/AsyncConnection.cc

```diff
@@ -977,9 +977,8 @@ ssize_t AsyncConnection::_process_connection()
                                 << " not " << peer_addr
                                 << " - presumably this is the same node!" << dendl;
           } else {
-            ldout(async_msgr->cct, 0) << __func__ << " connect claims to be "
-                                << paddr << " not " << peer_addr << " - wrong node!" << dendl;
-            goto fail;
+            ldout(async_msgr->cct, 20) << __func__ << " connect claims to be "
```

Comment:
should this check still be in place and look for either the public_addr or public_bind_addr? At a minimum I wonder if the previous behavior of failing should remain when not falling under the new scenario. And in the new scenario I don't think we would want to write the issue to the log since it's expected.